en/docs/apis/restapis/challenge-questions.yaml

```diff
@@ -0,0 +1,388 @@
+        Adds a new challenge question set to the system. A challenge question set can have any number of questions.
+
+          <b>Permission required:</b>
+            * /permission/admin/manage/identity/challenge/create
+      consumes:
+      - application/json
+      parameters:
+      - in: body
+        name: challengeSet
+        description: Challenge question set to add
+        schema:
+          type: array
+          items:
+            $ref: '#/definitions/ChallengeSet'
+      responses:
+        201:
+          $ref: '#/responses/Created'
+        400:
+          $ref: '#/responses/InvalidInput'
+        401:
+          $ref: '#/responses/Unauthorized'
+        409:
+          $ref: '#/responses/Conflict'
+        500:
+          $ref: '#/responses/ServerError'
+    get:
+      tags:
+        - admin
+      summary: Retrieve all the challenge questions.
+      operationId: searchChallenges
+      produces:
+      - application/json
+      description: |
+        Retrieve all the challenge questions in the system.
+
+          <b>Permission required:</b>
+            * /permission/admin/manage/identity/challenge/view
+      parameters:
+        - $ref: '#/parameters/localeQueryParam'
+        - $ref: '#/parameters/offsetQueryParam'
+        - $ref: '#/parameters/limitQueryParam'
+      responses:
+        200:
+          description: search results matching criteria
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/ChallengeSet'
+        401:
+          $ref: '#/responses/Unauthorized'
+        500:
+          $ref: '#/responses/ServerError'
+  /challenges/{challenge-set-id}:
+    get:
+      tags:
+        - admin
+      summary: Retrieve a challenge set.
+      operationId: getChallengeQuestionSet
+      produces:
+      - application/json
+      description: |
+        Retrieve the challenge questions in the system in a set identified by the challenge-set-id.
+
+          <b>Permission required:</b>
+            * /permission/admin/manage/identity/challenge/view
+      parameters:
+        - $ref: '#/parameters/questionSetIdPathParam'
+        - $ref: '#/parameters/localeQueryParam'
+        - $ref: '#/parameters/offsetQueryParam'
+        - $ref: '#/parameters/limitQueryParam'
+      responses:
+        200:
+          description: search results matching criteria
+          schema:
+            $ref: '#/definitions/ChallengeSet'
+        400:
+          $ref: '#/responses/InvalidInput'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+        500:
+          $ref: '#/responses/ServerError'
+    put:
+      tags:
+        - admin
+      summary: Update challenge questions of a set.
+      operationId: updateChallengeQuestionSet
+      description: |
+        Updates an existing challenge question set in the system. This will override the existing challenge questions in the set by new challenge questions.
```

Comment:
```suggestion
        Updates an existing challenge question set in the system. This will override the existing challenge questions in the set with new challenge questions.
```